src/core/Form/TimeInput/TimeInput.tsx

```diff
@@ -0,0 +1,300 @@
+   * not the blur FocusEvent like in other input components in this library!
+   */
+  onBlur?: (inputValue: string) => void;
+  /** Label for the input */
+  labelText: ReactNode;
+  /** Hides or shows the label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: LabelMode;
+  /** Hint text to be shown below the component */
+  hintText?: string;
+  /**
+   * `'default'` | `'error'`
+   *
+   * Status of the component. Error status creates a red border around the input.
+   * Always use a descriptive `statusText` with an error status.
+   * @default default
+   */
+  status?: InputStatus;
+  /**
+   * Type of the input
+   * @default text
+   */
+  /** HTML name attribute for the input */
+  name?: string;
+  /** Initial value for the input */
+  defaultValue?: string;
+  /** Controlled value */
+  value?: string;
+  /** Sets component's width to 100% of its parent */
+  fullWidth?: boolean;
+  /** Text to mark the field optional. Will be wrapped in parentheses and shown after `labelText` */
+  optionalText?: string;
+  /** Debounce time in milliseconds for onChange function. No debounce is applied if no value is given. */
+  debounce?: number;
+  /** Suomi.fi icon to be shown inside the input field */
+  icon?: ReactElement;
+  /** Tooltip component for the input's label */
+  tooltipComponent?: ReactElement;
+}
+
+const BaseTimeInput = (props: TimeInputProps) => {
+  const {
+    className,
+    labelText,
+    labelMode,
+    onChange: propOnChange,
+    onBlur: propOnBlur,
+    wrapperProps,
+    optionalText,
+    status,
+    statusText,
+    hintText,
+    id,
+    fullWidth,
+    icon,
+    value: controlledValue,
+    defaultValue,
+    forwardedRef,
+    debounce,
+    statusTextAriaLiveMode = 'assertive',
+    'aria-describedby': ariaDescribedBy,
+    tooltipComponent,
+    ...rest
+  } = props;
+  const [marginProps, passProps] = separateMarginProps(rest);
+  const marginStyle = spacingStyles(marginProps);
+
+  const [inputValue, setInputValue] = useState(defaultValue || '');
+
+  const inputRef = useRef<HTMLInputElement>(null);
+  const definedRef = forwardedRef || null;
+
+  // String is of type XX:YY or XX.YY where XX is 0-24 and YY is 0-59
+  const isValidTimeString = (timeStr: string) => {
+    if (timeStr.match(/^\d{1,2}.\d{2}$/) || timeStr.match(/^\d{1,2}:\d{2}$/)) {
+      const parts = timeStr.split(timeStr.includes('.') ? '.' : ':');
+      const hours = parseInt(parts[0], 10);
+      const minutes = parseInt(parts[1], 10);
+      if (hours >= 0 && hours < 25 && minutes >= 0 && minutes < 60) {
+        return true;
+      }
+    }
+
+    return false;
+  };
+
+  const handleOnBlur = () => {
+    let adjustedInputValue = '';
+    const inputValInt = parseInt(inputValue, 10);
+
+    // Handle automatic filling of 1 or 2 characters: 14 --> 14.00.
+    // Also remove leading zero from hours
+    if (
+      (inputValue.match(/^\d{1}$/) || inputValue.match(/^\d{2}$/)) &&
```

Comment:
Could this be simplified to just `(inputValue.match(/^\d{1,2}$/)`?

Reply:
Indeed! Changed it now